config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -72,19 +74,29 @@ public Optional<T> getMatchingData(RequestContext context, F filterInput, List<S
     return getMatchingObject(context, filterInput, sortInput).map(ConfigObject::getData);
   }
 
-  public ConfigsResponse<T> getMatchingObjectsWithTotalCount(
-      RequestContext context, F filterInput, List<S> sortInput, Pagination pagination) {
+  public ConfigsResponse<ContextualConfigObject<T>> getMatchingObjectsWithTotalCount(
```

Comment:
So in my earlier comment I was suggesting something more like
```suggestion
  public ConfigsResponse<T> getMatchingDataWithTotalCount(
```

Unless we actually need the `ContextualConfigObject` today, but those verbose versions of the APIs aren't really used much. We can always add the `ConfigsResponse<ContextualConfigObject<T>> getMatchingObjectsWithTotalCount` form if needed later to have the symmetry with the other APs.

Reply:
It will be 
`(ConfigsResponse<T>) getMatchingObjects(context, filterInput, sortInput, pagination, true);`

Reply:
It shouldn't require a cast - `getMatchingObjects` should be returning `ConfigsResponse<T>` instead of `ConfigsResponse<ContextualConfigObject<T>>` as it's currently declared. The impl would need to change.

Reply:
Understood now, added both objects and data api and updated accordingly